File: Version-1_Payload/src/modules/auxiliary/CmdResult.java
package modules.auxiliary;

/**
    Represents the result of a command execution to be sent
    back to the API.
 */
public class CmdResult {
    public String targetId;
    public String resultString;

    public CmdResult() {}

    public CmdResult(String targetId, String resultString) {
        this.targetId = targetId;
        this.resultString = resultString;
    }

    public CmdResult targetId(String id) {
        targetId = id;

        return this;
    }

    public CmdResult resultString(String resultString) {
        resultString = resultString;

        return this;
    }
}

File: api/Version-1_API/Model/CmdResult.cs
namespace Version_1_API.Model
{
    public class CmdResult
    {
        public string targetId { get; set; }
        public string? resultString { get; set; }

        public CmdResult() { }

        public CmdResult(string id, string result) { 
            targetId = id;
            resultString = result;
        }
    }
}
